### DISCUSSION TITLE
How to get lastId from args from graphql query

### DISCUSSION THREAD

**Discussion by rahXephonz** [prisma/prisma#12013] [Q&A] [answered]
Hi ,

Want ask again i have a trouble from get lastId from my query cannot return nullable from query.comments
Expectation I want to query a lastid if i input an argument from args query and want it works, but if i didnt input an arguments from args query i want it works fine too..

model prisma
```prisma
model Comment {
  id        Int      @id @default(autoincrement())
  text      String
  post      Post     @relation(fields: [postId], references: [id])
  user      User     @relation(fields: [userId], references: [id])
  userId    Int
  postId    Int
  createdAt DateTime @default(now())

  @@map(name: "comments")
}
```
my query typedefs
```ts
comments(limit: Int!, offset: Int!, postId: ID, lastId: ID): [Comment]!
```

my query resolvers
```ts
  comments: async (
    _: any,
    // this is args from query graphql
    { limit, offset, postId, lastId }: OpArgs,
    { db }: Context
  ) => {
    const getAllComments = await db.comment.findMany({
      where: {
        postId: +postId,
      },

      orderBy: [
        {
          createdAt: "asc",
        },
      ],

      include: { user: true },
      take: limit,
      skip: offset,
    });

    return getAllComments;
  },
```  

I hope someone help me.. I want learn a lot of prisma :)

**Comment by rahXephonz** [comment, marked as answer]
Oops.. I think i got this answer by reading documentations again and adding gt in where a data want to be querying
```ts
      where: {
        postId: +postId,
       // by this
        id: +lastId ? { gt: +lastId } : undefined,
      },
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
